### PR TITLE
Print line number and offending entry on plugins parse failure

### LINKED ISSUE #435: Print load order line and modname on failed plugins.txt parsing
```
[1.4][Skyrim Special Edition] Load order location error: System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Mutagen.Bethesda.Plugins.ModKey.TryFromName(ReadOnlySpan`1 str, ModType type, ModKey& modKey, String& errorReason)
   at Mutagen.Bethesda.Plugins.ModKey.TryFromNameAndExtension(ReadOnlySpan`1 str, ModKey& modKey, String& errorReason)
   at Mutagen.Bethesda.Plugins.ModKey.TryFromNameAndExtension(ReadOnlySpan`1 str, ModKey& modKey)
   at Mutagen.Bethesda.Plugins.Order.LoadOrderListing.TryFromString(ReadOnlySpan`1 str, Boolean enabledMarkerProcessing, LoadOrderListing& listing)
   at Mutagen.Bethesda.Plugins.Order.LoadOrderListing.FromString(ReadOnlySpan`1 str, Boolean enabledMarkerProcessing)
   at Mutagen.Bethesda.Plugins.Order.DI.LoadOrderListingParser.FromString(ReadOnlySpan`1 str)
   at Mutagen.Bethesda.Plugins.Order.DI.PluginListingsParser.Parse(Stream stream)+MoveNext()
```

### PR DESCRIPTION
**Pull request by studioph**
## Changes
- Adds new test case for failure plugin parsing to `Mutagen.Bethesda.Core.UnitTests/Plugins/Order/PluginListingsParserTests.cs`
- Updates `PluginListingsParser.Parse` to track the current line number for inclusion in error message

Closes #435

### CODE REVIEW

**Comment by studioph** [comment]
Dangit - looks like there's some frivolous whitespace changes that I need to fix

File: Mutagen.Bethesda.Core.UnitTests/Plugins/Order/PluginListingsParserTests.cs [hunk 1]
```diff
@@ -73,4 +73,22 @@ public void CommentTrimming()
...
+                new HasEnabledMarkersInjection(true)));
+
+        parser.Invoking(p => p.Parse(GetStream(
+@"*ModA.esm
+ModB.esp
+*Malformed"))
+            .ToList())
+            .Should()
+            .Throw<InvalidDataException>()
+            .WithMessage("Load order file had malformed entry at line 3: \"*Malformed\"");
```
Comment (Noggog) [inline Mutagen.Bethesda.Core.UnitTests/Plugins/Order/PluginListingsParserTests.cs:92]: Ah, didn't know this was available FluentAssertion API!  TIL

**Review by Noggog — APPROVED** [review]

**Comment by Noggog** [comment]
> Dangit - looks like there's some frivolous whitespace changes that I need to fix

Let me know when your changes are in.  Looks great overall!

**Comment by studioph** [comment]
Yea I forgot to set `git config core.autocrlf` I think...

EDIT: Actually it's not CRLF - there's just a bunch of spaces on those lines that got removed in formatting

**Comment by Noggog** [comment]
Not a huge deal or anything to go back and fix, but I think you did some odd cherry pick operation that duplicated the `dev` commits THEN merged in `dev` after that
image

Not a big deal but figured id mention for next time

**Comment by studioph** [comment]
> Not a huge deal or anything to go back and fix, but I think you did some odd cherry pick operation that duplicated the `dev` commits THEN merged in `dev` after that image
> 
> Not a big deal but figured id mention for next time

Huh, I did do a rebase at one point with the intention of keeping everything "clean" but I guess that just made it worse. I worry I may never master git...

I'm happy to try and fix the history so it doesn't screw up anything else

**Comment by studioph** [comment]
@Noggog I think I was able to fix it and remove the duplicate commits - the network graph looks more normal now (at least on my end)

**Review by Noggog — APPROVED** [review]

**Pull request merged**